### ISSUE TITLE
Hunter's quest "You already control a tamed creature" error

### ISSUE THREAD

**Issue by MsgLosers** [dalaranwow/dalaran-wow#5674]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Upon completing the hunter pet taming quest chain, the tamed pet does not disassociate from the hunter (but all UI for it disappears) so you cannot immediately tame the next pet without log-out and log-in cycle...

EDIT: During my attempt at part 2 & part 3, the UI remained up (but the Dismiss option does not work, so this is still bugged).

**Expected behaviour**: Tell us what should happen instead
Upon completing each quest, either keep the UI up so we can manually dismiss the pet as in vanilla, or automatically dismiss the pet for the player.

**Steps to reproduce the problem**:

1. Accept hunter quest.
2. Tame first pet.
3. Deliver quest (~~this loses your temporary pet and ui actions~~) and accept next part
4. Attempt to tame next pet: Fails upon 100% tame bar with error message similar to "you already control a tamed creature..." 

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Lothloryen** [comment]
Hello there,

Dismiss the pet you did obtain while doing the quest first, please.

Regards.
